src/main/java/com/bookshop/dto/CreateBookRequestDto.java

```diff
@@ -1,11 +1,21 @@
 package com.bookshop.dto;
 
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Pattern;
+import jakarta.validation.constraints.Positive;
 import java.math.BigDecimal;
 
 public record CreateBookRequestDto(
+        @NotBlank
         String title,
+        @NotBlank
         String author,
+        @NotBlank
+        @Pattern(regexp = "^\\d{3}-\\d{10}$")
```

Comment:
I assume that `@NotBlank` is redundant here, since blank string should not match the regex

double check that :)